scripts/ci-cd/input_validator.py

```
@@ -0,0 +1,139 @@
+
+    @staticmethod
+    def validate_pr_number(pr_number: Any) -> int:
+        """Validate PR number is a positive integer"""
+        try:
+            pr_num = int(pr_number)
+            if pr_num <= 0:
+                raise ValueError("PR number must be positive")
+            if pr_num > 999999:  # Reasonable upper limit
+                raise ValueError("PR number seems unreasonably large")
+            return pr_num
+        except (TypeError, ValueError) as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)